advisories/2024-09-25_cometbft-bls12381-key-validation.md

```diff
@@ -0,0 +1,128 @@
+# Missing BLS12-381 Public Key Validation
+
+## Description
+
+CometBFT latest available release is v1.0.0-rc1.0.20240805092115-3b2c5d9e1843, which introduces support for BLS signatures over the BLS12-381 curve. In this release, instantiation of BLS public keys from byte arrays is done via `github.com/cosmos/crypto v0.0.0-20240309083813-82ed2537802e`, a clone of the Prysm's wrapper over BLST, as it can be seen at: https://github.com/cometbft/cometbft/blob/86da0027d878707365c16b124b77892ca5212fe1/crypto/bls12381/key_bls12381.go#L127 The `PublicKeyFromBytes` function performs the G1 subgroup and infinity check with a call to BLST's `KeyValidate`: https://github.com/cosmos/crypto/blob/bb8c5deb91b3a722e145c4d9c6d06c6158d23dfe/curves/bls12381/pubkey.go#L64
```

Comment:
I'm not sure if this is the actual current release. The tense of the advisory should be past-tense if this was the latest release at the time of writing (you could also include this). My suggestion is to have it read something like this:

"CometBFT's latest release, at the time of writing, was `version`. This release introduced support for BLS signatures..." and continue to describe what was, followed by the bug introduced on the main branch on September 11.

Reply:
Additionally, links to code should be converted to hyperlinks. Makes for easier reading.